### DISCUSSION TITLE
Using tsvector operators

### DISCUSSION THREAD

**Discussion by sumeetkh** [hasura/graphql-engine#7505] [Question] [answered]
Hi,

I am trying to learn how to use tsvector operators in a graphql query, but not able to find documentation about it. Here's the corresponding sql query I am trying to run.

select * from table where text_search_vector @@ plainto_tsquery('search_string')

Is this doable with Hasura operators?

Thank you!

**Comment by BenoitRanque** [comment, marked as answer]
[here](https://hasura.io/docs/latest/graphql/core/api-reference/syntax-defs.html#operator) is a list of supported postgres operators and their equivalent in graphql. 
There is no currently available way to use tsvector with the api automatically provided by hasura.
This leaves you with a few options, the simplest of which is most likely [table computed fields](https://hasura.io/docs/latest/graphql/core/databases/postgres/schema/computed-fields.html#table-computed-fields).
There is an open RFC for parametrized queries which will offer a more ergonomic solution in the future.
You can find it [here](https://github.com/hasura/graphql-engine/issues/7473)

**Comment by sumeetkh** [comment]
Thanks for you response Benoit.

I looked into custom defined functions option - but it only solves the most basic use cases. Based on playing around with it, I am only able to send text arguments to the custom function and the graphql api does not interpret a boolean tsquery.

To get this to work on a general query would require a lot of string wrangling to get the query into a shape that would work. 

I think if the RFC is implemented, it would certainly help for these use cases.

**Reply by BenoitRanque** [reply]
Note that even the RFC will _not_ allow you to "interpret a boolean tsquery". That would effectively be SQL injection which is a security issue.
In both cases you would send one or more text arguments to search with, and craft an SQL query using those arguments to return your filtered data.

**Comment by sumeetkh** [comment]
Got it, thanks Benoit.  In that case, in the absence of first class support for text search operators https://www.postgresql.org/docs/10/functions-textsearch.html in the query language, doing non-trivial search over the graphql api becomes really hard.

E.g. a interpreting a boolean expression with multiple clauses in the custom function,  "term1" && ("term2" || "term3")

**Reply by BenoitRanque** [reply]
I'm not sure I understand. You can pass multiple arguments to the custom function. You could even pass arrays if the quantity of arguments of X type is unknown.

**Reply by sumeetkh** [reply]
Yeah, I found about variadic arguments support in postgres custom functions - so the use case is doable for now. It still requires a fair bit of manipulation that I'd have liked to avoid, but it works for now.
